packages/single-spa-welcome/src/welcome.js

```diff
@@ -1,7 +1,24 @@
-import React from "react";
+import React, { useState } from "react";
 import "./welcome.css";
 
 export default function Root(props) {
+  const isUsingSingleSpaLayout = useState(() => {
+    // Search the dom for a single-spa-router element
+    // While this won't work for SSR, the goal of this app is to help
+    // beginners, which likely won't use SSR before running their
+    // first root-config.
+    const templates = Array.prototype.slice.call(
```

Comment:
Maybe easiest to check this way?

```js
return Boolean(document.querySelector('template[id="single-spa-layout"]'))
```

Reply:
Yea I debated whether to do that simple check. Its highly unlikely but still possible that someone changes that id before running their root-config so it would show the wrong instructions. I would just be annoyed if someone opened up an issue caused by that, and I would think a beginner would be frustrated if the instructions don't match up to their code. Maybe I should just create collapsible sections and the user should expand the section with specific instructions themselves?

Reply:
I'm okay with the longer implementation, too, if you'd prefer it. The longer implementation is more correct for that edge case.

Reply:
Let me think about it. Some of that information can/should be duplicated as part of the ["Getting Started" update](https://github.com/single-spa/single-spa.js.org/issues/335) I suggested and have begun on, and that might help me decide which strategy is best.